src/main/java/org/truffleruby/language/dispatch/DispatchNode.java

```diff
@@ -128,7 +128,8 @@ public Object execute(VirtualFrame frame, Object receiver, String methodName, Ru
                     return MISSING;
                 case CALL_METHOD_MISSING:
                     // Both branches implicitly profile through lazy node creation
-                    if (metaclass == getContext().getCoreLibrary().truffleInteropForeignClass) {
+                    if (metaclass == getContext().getCoreLibrary().polyglotForeignArrayClass ||
+                            metaclass == getContext().getCoreLibrary().polyglotForeignObjectClass) {
                         return callForeign(receiver, methodName, block, arguments);
                     } else {
```

Comment:
@eregon How can I check the metaclass inheritance so that I can check only against polyglotForeignObjectClass?

Reply:
You can use a `IsANode` for that.